packages/react-native-audio-api/common/cpp/audioapi/events/AudioEventHandlerRegistry.cpp

```diff
@@ -0,0 +1,109 @@
+#include <audioapi/events/AudioEventHandlerRegistry.h>
+
+namespace audioapi {
+
+AudioEventHandlerRegistry::AudioEventHandlerRegistry(
+    jsi::Runtime *runtime,
+    const std::shared_ptr<react::CallInvoker> &callInvoker) {
+  runtime_ = runtime;
+  callInvoker_ = callInvoker;
+
+  for (const auto &eventName : SYSTEM_EVENT_NAMES) {
+    eventHandlers_[std::string(eventName)] = {};
+  }
+
+  for (const auto &eventName : AUDIO_API_EVENT_NAMES) {
+    eventHandlers_[std::string(eventName)] = {};
+  }
+}
+
+AudioEventHandlerRegistry::~AudioEventHandlerRegistry() {
+  eventHandlers_.clear();
+}
+
+uint64_t AudioEventHandlerRegistry::registerHandler(
+    const std::string &eventName,
+    const std::shared_ptr<jsi::Function> &handler) {
+  static uint64_t LISTENER_ID = 0;
+  LISTENER_ID++;
+
+  eventHandlers_[eventName][LISTENER_ID] = handler;
```

Comment:
```suggestion
  eventHandlers_[eventName][LISTENER_ID++] = handler;
```
why not like that, also why starting count from 1?